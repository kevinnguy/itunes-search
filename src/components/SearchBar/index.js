import React from 'react';
import {
  TouchableOpacity,
  StyleSheet,
  TextInput,
  Text,
  View,
} from 'react-native';

import Loading from '../Loading/index.js';

export default function SearchBar({
  isLoading,
  style,
  onChangeText,
  onSubmit,
  onPress,
}) {
  return (
    <View style={[styles.container, style]}>
      <TextInput
        autoFocus
        style={styles.textInput}
        onChangeText={onChangeText}
        onSubmitEditing={onSubmit}
        numberOfLines={1}
        autoCorrect={false}
        placeholder={'Search artists, songs, albums'}
      />
      {isLoading ? (
        <Loading style={styles.loading} />
      ) : (
        <TouchableOpacity style={styles.searchButton} onPress={onPress}>
          <Text style={styles.searchButtonText}>{'Search'}</Text>
        </TouchableOpacity>
      )}
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    height: 60,
    alignItems: 'center',
    paddingHorizontal: 20,
    borderBottomColor: '#888',
    borderBottomWidth: StyleSheet.hairlineWidth,
  },
  textInput: {
    flex: 1,
    fontSize: 20,
  },
  loading: {
    marginLeft: 10,
  },
  searchButton: {
    borderColor: 'purple',
    backgroundColor: 'purple',
    borderRadius: 10,
    borderWidth: StyleSheet.hairlineWidth,
    padding: 10,
  },
  searchButtonText: {
    color: 'white',
  },
});
