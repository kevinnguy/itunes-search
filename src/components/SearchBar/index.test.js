import React from 'react';
import renderer from 'react-test-renderer';

import SearchBar from './index';

test('renders correctly', () => {
  const tree = renderer.create(<SearchBar />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders correctly when loading', () => {
  const tree = renderer.create(<SearchBar isLoading />).toJSON();
  expect(tree).toMatchSnapshot();
});
