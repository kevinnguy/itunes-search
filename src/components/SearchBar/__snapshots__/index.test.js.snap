// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderBottomColor": "#888",
        "borderBottomWidth": 0.5,
        "flexDirection": "row",
        "height": 60,
        "paddingHorizontal": 20,
      },
      undefined,
    ]
  }
>
  <TextInput
    allowFontScaling={true}
    autoCorrect={false}
    autoFocus={true}
    numberOfLines={1}
    placeholder="Search artists, songs, albums"
    rejectResponderTermination={true}
    style={
      Object {
        "flex": 1,
        "fontSize": 20,
      }
    }
    underlineColorAndroid="transparent"
  />
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "purple",
        "borderColor": "purple",
        "borderRadius": 10,
        "borderWidth": 0.5,
        "opacity": 1,
        "padding": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
        }
      }
    >
      Search
    </Text>
  </View>
</View>
`;

exports[`renders correctly when loading 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderBottomColor": "#888",
        "borderBottomWidth": 0.5,
        "flexDirection": "row",
        "height": 60,
        "paddingHorizontal": 20,
      },
      undefined,
    ]
  }
>
  <TextInput
    allowFontScaling={true}
    autoCorrect={false}
    autoFocus={true}
    numberOfLines={1}
    placeholder="Search artists, songs, albums"
    rejectResponderTermination={true}
    style={
      Object {
        "flex": 1,
        "fontSize": 20,
      }
    }
    underlineColorAndroid="transparent"
  />
  <Image
    source={
      Object {
        "uri": "https://www.pngitem.com/pimgs/m/235-2350144_load-map-bar-ios-5s-iphone-progress-clipart.png",
      }
    }
    style={
      Array [
        Object {
          "height": 40,
          "width": 40,
        },
        Object {
          "marginLeft": 10,
        },
      ]
    }
  />
</View>
`;
